src/controllers/levelcontroller.js

```diff
@@ -41,6 +41,8 @@ class LevelController extends Controller
             showConfirmButton: false
         });
 
+        (new FactoryController)._updateFactory();
```

Comment:
Okay, I have an idea, but it'll take longer to explain than for me to do once this is merged. This can stay (it wasn't wrong after all).